### PR TITLE
Switch to Jest

### LINKED ISSUE #74: Move testing framework over to JEST with typescript
Current testing framework is ultra annoying since it tests the `dist` folder.   Let's switch to JEST

Maybe with https://github.com/skovhus/jest-codemods  ???

This will bring our test coverage to a real number.

### PR DESCRIPTION
**Pull request by GantMan**
* Convert AVA to Jest tests with code mods
* Change coverage reporter to work with Jest
* Change everything to TypeScript!
* Real COVERAGE!!!!
---
Closes #74

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,8 +34,8 @@
     "docsify-cli": "^4.1.10",
     "execa": "^0.8.0",
     "jest": "^21.2.1",
-    "jest-cli": "^21.2.1",
```
Comment (skellock) [inline package.json]: Heads up here @GantMan

File: package.json [hunk 2]
```diff
@@ -33,8 +34,8 @@
...
     "np": "^2.16.0",
+    "nyc": "^11.2.1",
```
Comment (skellock) [inline package.json]: I think `nyc` is handled automatically isn't it by Jest?

File: package.json (hunk 1 above)
Comment (GantMan) [reply package.json]: I don't think you're supposed to have jest and jest-cli

File: package.json (hunk 2 above)
Comment (GantMan) [reply package.json]: it is.  I don't know how this snuck back in.

File: package.json (hunk 1 above)
Comment (skellock) [reply package.json]: Ahhh.  My brain read this as you removing jest. Oops.

**Pull request merged**